core/src/main/scala/kafka/coordinator/transaction/ProducerIdManager.scala

```diff
@@ -236,7 +236,7 @@ class RPCProducerIdManager(brokerId: Int,
   private[transaction] def handleTimeout(): Unit = {
     warn("Timed out when requesting AllocateProducerIds from the controller.")
     requestInFlight.set(false)
-    nextProducerIdBlock.put(Failure(Errors.REQUEST_TIMED_OUT.exception))
+    nextProducerIdBlock.put(Failure(Errors.COORDINATOR_LOAD_IN_PROGRESS.exception))
```

Comment:
I wonder why we do this. If we don't send anything back, we still get the timeout after `maxWaitMs`. The odd thing is that the timeout might occur in the middle of the wait for `maxWaitMs`, so the actual time we wait might be less than that. I wonder if we can just get rid of this?

Reply:
I think it so we return the error immediately instead of waiting for the poll. 
We also only send one request -- once we get this response we will drop to the block were we poll and get this error response out.